python/tk_multi_workfiles/my_tasks/my_tasks_model.py

```diff
@@ -13,19 +13,27 @@
 """
 
 import sgtk
-from sgtk.platform.qt import QtGui
+from sgtk.platform.qt import QtCore, QtGui
 
 from ..util import resolve_filters
 from ..entity_models import ShotgunExtendedEntityModel
 
+delegates = sgtk.platform.import_framework("tk-framework-qtwidgets", "delegates")
+ViewItemRolesMixin = delegates.ViewItemRolesMixin
 
-class MyTasksModel(ShotgunExtendedEntityModel):
+
+class MyTasksModel(ShotgunExtendedEntityModel, ViewItemRolesMixin):
     """
     Specialisation of the Shotgun entity model that represents a single users tasks.  Note that we derive
     from the Shotgun entity model so that we have access to the entity icons it provides.  These are used
-    later by the MyTaskItemDelegate when rending a widget for a task in the My Tasks view.
+    later by the delegate when rendering a task item in the My Tasks view.
     """
 
+    _BASE_ROLE = QtCore.Qt.UserRole + 32
+    (NEXT_AVAILABLE_ROLE,) = range(_BASE_ROLE, _BASE_ROLE + 1)
```

Comment:
What does 32 mean? Is it possible to extract this magic value to a constant?

Reply:
This is a Toolkit legacy thing...

Basically the purpose here is to ensure we use model item data roles that Qt does not use. Qt provides `QtCore.Qt.UserRole` as the first model item data role that applications can use to add their own data roles. Most data models in Toolkit derive from `ShotgunModel`, which derives from the `ShotgunQueryModel`; both of these models define some custom roles (for example [here](https://github.com/shotgunsoftware/tk-framework-shotgunutils/blob/57935971acd686210db540b5bd8ef842b373d321/python/shotgun_model/shotgun_query_model.py#L102-L108)). So I believe the arbitrary number of "32" was picked to ensure any model deriving from these models, would not override the base model custom roles. While this method works, it is not a bullet proof since if the base models needed to define more custom roles, and exceeded 32, then we would have conflicts.

So all that to say, here I am following the precedent set by other Toolkit apps, but it would be ideal if the shotgun models in `tk-framework-shotgunutils` were modified to provide a more "official" model role that is safe to use by models that inherit the base shotgun models (similarly to how Qt provides `UserRole`).

I am happy to implement the proposed change, but it extends the scope of this ticket. Do we want to include this, or possibly create a new ticket for it?